src/systemctl/systemctl.c

```diff
@@ -7264,7 +7268,58 @@ static int create_edit_temp_file(const char *new_path, const char *original_path
         if (r < 0)
                 return log_error_errno(r, "Failed to create directories for \"%s\": %m", new_path);
 
-        r = copy_file(original_path, t, 0, 0644, 0, 0, COPY_REFLINK);
+        if (original_path) {
+                r = copy_file(original_path, t, 0, 0644, 0, 0, COPY_REFLINK);
+        } else {
+                _cleanup_free_ char *new_contents = NULL;
+                _cleanup_fclose_ FILE *f = NULL;
+                char **path;
+                size_t size;
+
+                f = fopen(t, "we");
+                if (!f)
+                        return log_error_errno(errno, "Failed to open \"%s\": %m", t);
+
+                r = fchmod(fileno(f), 0644);
+                if (r < 0)
+                        return log_error_errno(errno, "Failed to change mode of \"%s\": %m", t);
+
+                r = read_full_file(new_path, &new_contents, &size);
```

Comment:
don't you want to copy original_path in here?

Reply:
In this branch the `new_path` is the override unit being edited, so this is to read the current contents of that file so we can put it in the temp file.
`original_path` is only available if we're overriding a single file, and that's handled by branch above.
'original' in this case is trying to indicate these are unaltered vendor unit files - might be a bit of a weird naming scheme, but I decided to keep the names of the existing variables.